app/src/main/java/fr/free/nrw/commons/explore/depictions/SearchDepictionsFragment.java

```diff
@@ -0,0 +1,238 @@
+import android.content.res.Configuration;
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ProgressBar;
+import android.widget.TextView;
+
+import androidx.recyclerview.widget.GridLayoutManager;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.pedrogomez.renderers.RVRendererAdapter;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Locale;
+import java.util.concurrent.TimeUnit;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+import fr.free.nrw.commons.R;
+import fr.free.nrw.commons.depictions.DepictedImagesActivity;
+import fr.free.nrw.commons.di.CommonsDaggerSupportFragment;
+import fr.free.nrw.commons.explore.recentsearches.RecentSearch;
+import fr.free.nrw.commons.explore.recentsearches.RecentSearchesDao;
+import fr.free.nrw.commons.kvstore.JsonKvStore;
+import fr.free.nrw.commons.mwapi.MediaWikiApi;
+import fr.free.nrw.commons.upload.structure.depicts.DepictedItem;
+import fr.free.nrw.commons.utils.NetworkUtils;
+import fr.free.nrw.commons.utils.ViewUtil;
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.schedulers.Schedulers;
+import timber.log.Timber;
+
+import static android.view.View.GONE;
+import static android.view.View.VISIBLE;
+
+public class SearchDepictionsFragment extends CommonsDaggerSupportFragment {
+    private static int TIMEOUT_SECONDS = 15;
+
+    @BindView(R.id.imagesListBox)
+    RecyclerView depictionsRecyclerView;
+    @BindView(R.id.imageSearchInProgress)
+    ProgressBar progressBar;
+    @BindView(R.id.imagesNotFound)
+    TextView depictionNotFound;
+    String query;
+    @BindView(R.id.bottomProgressBar)
+    ProgressBar bottomProgressBar;
+    boolean isLoadingDepictions;
+    @Inject
+    RecentSearchesDao recentSearchesDao;
+    @Inject
+    MediaWikiApi mwApi;
+    @Inject
+    DepictsClient depictsClient;
+    @Inject
+    @Named("default_preferences")
+    JsonKvStore basicKvStore;
+    private RVRendererAdapter<DepictedItem> depictionsAdapter;
+    private List<DepictedItem> queryList = new ArrayList<>();
+
+    private final SearchDepictionsAdapterFactory adapterFactory = new SearchDepictionsAdapterFactory(item -> {
+        // Called on Click of a individual depicted Item
+        // Open Depiction Details activity
+        DepictedImagesActivity.startYourself(getContext(), item);
+        saveQuery(query);
+    });
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        View rootView = inflater.inflate(R.layout.fragment_browse_image, container, false);
+        ButterKnife.bind(this, rootView);
+        if (getActivity().getResources().getConfiguration().orientation == Configuration.ORIENTATION_PORTRAIT) {
```

Comment:
Why getActivity(), can we not used the fragment context ?

Reply:
that's fine, replaced with `getContext()` in the next commit

Reply:
added in the next commit